woocommerce/woocommerce-gateway-stripe
# Add E2E Woo Subscriptions setup

Closes #2521 

## Changes proposed in this Pull Request:

This PR streamlines the E2E testing process by automating the setup of WC Subscriptions in the environment. This eliminates the need for manual installation, ensuring smoother and more efficient testing.

## Testing instructions

1. Setup a new E2E environment on a JN site. Following the instructions [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/tests/e2e/README.md).
2. There's a new variable in the `local.env` env file, `GITHUB_TOKEN`. This token should have access to the WC Subscriptions repository. The plugin will be downloaded from the latest release.
3. After running `npm run test:e2e-setup -- --base_url=SOME_URL_HERE`, manually check that the subscriptions plugin was installed on the test site and that it is working.
4. Run all the E2E tests for subscriptions (`npm run test:e2e -- --base_url=SOME_URL_HERE --grep @subscriptions`) and ensure they're working and that you didn't have to do any setup on WC Subscriptions.



---


**Post merge**
